### DISCUSSION TITLE
CommandItem not updating on asynchronous changes

### DISCUSSION THREAD

**Discussion by ImamJanjua** [shadcn-ui/ui#3862] [General]
Hi, as u can see in the video i am trying to implement a component which on input change gets the addres predictions from server. Its using PopOver from shadcn for the for the popover.

Now the thing is that when the first adresses arrive than it will render it but on chnage of the input value, it gets the new predictions but it will not render them?

https://github.com/shadcn-ui/ui/assets/137432044/b571bbde-c29a-463a-a1d5-db994b5d76dd

```typescript
"use client";

import * as React from "react";
import { useDebounce } from "@/hooks/useDebounce";

import { Button } from "@/components/ui/Button";
import {
  Command,
  CommandEmpty,
  CommandList,
  CommandGroup,
  CommandInput,
  CommandItem,
} from "@/components/ui/Command";
import {
  Popover,
  PopoverContent,
  PopoverTrigger,
} from "@/components/ui/Popover";
import { Check, ChevronsUpDown } from "lucide-react";


export function ComboboxDemo() {
  const [open, setOpen] = React.useState(false);
  const [value, setValue] = React.useState("");
  const [selectedValue, setSelectedValue] = React.useState("");

  const [options, setOptions] = React.useState([] as any[]);
  const debouncedValue = useDebounce(value);

  React.useEffect(() => {
    async function handleValueChange() {
      // fetch needed data
      setOptions(results...);
    }
    handleValueChange();
  }, [debouncedValue]);

  return (
    <Popover open={open} onOpenChange={setOpen}>
      <PopoverTrigger asChild>
        <Button variant="outline" className="w-[200px] justify-between">
          {selectedValue
            ? options.find((option) => option.value === value)?.label
            : "Select framework..."}
          <ChevronsUpDown className="ml-2 h-4 w-4 shrink-0 opacity-50" />
        </Button>
      </PopoverTrigger>
      <PopoverContent className="p-0">
        <Command>
          <CommandInput
            value={value}
            onValueChange={setValue}
            placeholder="Search framework..."
          />
          <CommandList>
            <CommandEmpty>No framework found.</CommandEmpty>

            <CommandGroup>
              {options.map((option) => (
                <CommandItem
                  key={option.value}
                  value={option.value}
                  onSelect={(currentValue) => {
                    setSelectedValue(
                      currentValue === selectedValue ? "" : currentValue,
                    );
                    setOpen(false);
                  }}
                >
                  <Check
                    className={cn(
                      "mr-2 h-4 w-4",
                      selectedValue === option.value
                        ? "opacity-100"
                        : "opacity-0",
                    )}
                  />
                  {option.label}
                </CommandItem>
              ))}
            </CommandGroup>
          </CommandList>
        </Command>
      </PopoverContent>
    </Popover>
  );
}

export default ComboboxDemo;

```

**Comment by fitim63** [comment]
Passing `shouldFilter={false}` to the Command component should fix this.

**Reply by mdhvg** [reply]
This was such a lifesaver. Thank you so much.

**Reply by Maraoui21** [reply]
thank you soooooooooo much saved my time

**Reply by AnsariYasirArfat** [reply]
Thank you so much, bro!

**Reply by vuongmt2000** [reply]
nice!!!

**Reply by dpiyumal2319** [reply]
lUV YA

**Reply by AdrianMaj** [reply]
Thanks! This fixed my issue

**Reply by jflerime314** [reply]
Thank you. I've been trying to figure out the issue for more than a couple hours.

**Reply by thevideopilot** [reply]
Please, can someone help with where to put the "shouldFilter={false}"? I've tried putting it on the CommandItem but typecript gives an error
![Screenshot 2025-02-23 at 14 44 00](https://github.com/user-attachments/assets/0cef1ae2-9dac-473e-8cd9-97e992f3bd19)

**Reply by thevideopilot** [reply]
> Please, can someone help with where to put the "shouldFilter={false}"? I've tried putting it on the CommandItem but typecript gives an error ![Screenshot 2025-02-23 at 14 44 00](https://private-user-images.githubusercontent.com/50434578/415995413-0cef1ae2-9dac-473e-8cd9-97e992f3bd19.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAzMjIxNjUsIm5iZiI6MTc0MDMyMTg2NSwicGF0aCI6Ii81MDQzNDU3OC80MTU5OTU0MTMtMGNlZjFhZTItOWRhYy00NzNlLThjZDktOTdlOTkyZjNiZDE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjIzVDE0NDQyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyYzgxMTIwOTBhMTlmZGEyZGUyOTZjZTg0OTJkMGJjZDdlYzljNGQ2MjI3NWQ3MmM4ZTZlNDRlOTg1YWRjODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DZvxkRrlfkC7gRwrmqngGxN1O8b6C3--Z9YIINIwsUg)

Scratch this! Figured it out. It should be on the Command component (the first one)!

**Reply by hongdatchy** [reply]
@fitim63 Thank you, I spent 2 hours with chatGpt but only your answer could save my day!
